src/openms/source/ANALYSIS/ID/FragmentIndexScorer.cpp

```diff
@@ -284,13 +284,25 @@ namespace OpenMS
   }
 */
 
-  void FragmentIndexScorer::trimHits(OpenMS::FragmentIndexScorer::SpectrumMatchesTopN& init_hits)
+  void FragmentIndexScorer::trimHits(OpenMS::FragmentIndexScorer::SpectrumMatchesTopN& init_hits) const
   {
-    buildKMinHeap<SpectrumMatch, uint32_t>(init_hits.hits_, max_processed_hits_, [](SpectrumMatch a) { return a.num_matched_; });
     if (max_processed_hits_ < init_hits.hits_.size())
     {
+      std::partial_sort(init_hits.hits_.begin(), init_hits.hits_.begin() + max_processed_hits_, init_hits.hits_.end(), [](SpectrumMatch a, SpectrumMatch b){
+        return a.num_matched_ > b.num_matched_;
+      });
+
       init_hits.hits_.resize(max_processed_hits_);
     }
+    for(auto hit_iter = init_hits.hits_.begin(); hit_iter != init_hits.hits_.end();){
+      if(hit_iter->num_matched_ < min_matched_peaks_)
+      {
+        init_hits.hits_.erase(hit_iter);
```

Comment:
I think this code can be made much faster!!!

e.g. iterate from rbegin() (note: from the right) until we find the first entry that has hit_iter->num_matched_ >= min_matched_peaks_
and then just do a resize to include this position.
No erasing needed. Every erase needs to relocate all elements after the moved element.